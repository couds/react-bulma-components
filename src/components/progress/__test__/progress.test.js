import React from 'react';
import renderer from 'react-test-renderer';
import Progress from '..';
import CONSTANTS from '../../../constants';

describe('Progress component', () => {
  it('Should exist', () => {
    expect(Progress).toMatchSnapshot();
  });
  it('Should have notification classname', () => {
    const component = renderer.create(<Progress value={30} max={100} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(<Progress value={30} max={100} className="other-class this-is-a-test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(<Progress value={30} max={100} style={{ width: 250 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be Large', () => {
    const component = renderer.create(<Progress value={30} max={100} size="large" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  [null]
    .concat(Object.keys(CONSTANTS.COLORS).map(key => CONSTANTS.COLORS[key]))
    .map(color => (
      it(`Should use use color ${color}`, () => {
        const component = renderer.create(<Progress value={30} max={100} color={color} />);
        expect(component.toJSON()).toMatchSnapshot();
      })
    ));
});
