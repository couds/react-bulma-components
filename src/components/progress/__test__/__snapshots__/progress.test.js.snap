// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component Should be Large 1`] = `
<progress
  className="progress is-large"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should concat classname in props with Bulma classname 1`] = `
<progress
  className="progress other-class this-is-a-test"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should exist 1`] = `[Function]`;

exports[`Progress component Should have notification classname 1`] = `
<progress
  className="progress"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use inline styles 1`] = `
<progress
  className="progress"
  max={100}
  style={
    Object {
      "width": 250,
    }
  }
  value={30}
/>
`;

exports[`Progress component Should use use color danger 1`] = `
<progress
  className="progress is-danger"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color dark 1`] = `
<progress
  className="progress is-dark"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color info 1`] = `
<progress
  className="progress is-info"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color link 1`] = `
<progress
  className="progress is-link"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color null 1`] = `
<progress
  className="progress"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color primary 1`] = `
<progress
  className="progress is-primary"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color success 1`] = `
<progress
  className="progress is-success"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color text 1`] = `
<progress
  className="progress is-text"
  max={100}
  value={30}
/>
`;

exports[`Progress component Should use use color warning 1`] = `
<progress
  className="progress is-warning"
  max={100}
  value={30}
/>
`;
